### PR TITLE
Adding VolumeSnapshot CRDs and snapshot controller installation For GA

### LINKED ISSUE #328: Adding VolumeSnapshot CRDs and snapshot controller installation For GA
While installing VolumeSnapshot CRDs in snapshot controller installation, the Snapshot controller uses the version 2.0.1 and accordingly crd for version v1beta1 is applied. For User Using cluster v1.20 and above V1 version Crd should be installed , therefore adding steps for GA snapshot controller.

Reference : https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/docs/deploy-1.17-and-later.md#volumesnapshot-crds-and-snapshot-controller-installation

### PR DESCRIPTION
**Pull request by Kartik494**
**What type of PR is this?**
> /kind bug

Fixes #328

### CODE REVIEW

**Comment by Kartik494** [comment]
@xing-yang Hi , i have done the changes accordingly could You please take a look on this.
Thanks

**Comment by Kartik494** [comment]
Added a Commit for modify template of [ inline-ephemeral-support](https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/docs/example-ephemeral.md#inline-ephemeral-support)

**Comment by Kartik494** [comment]
@xing-yang Could you please take a look on this.
Thanks

**Comment by Kartik494** [comment]
Resolved the merge conflict. @andyzhangx @Jiawei0227 @xing-yang PTAL

File: README.md [hunk 1]
```diff
@@ -21,7 +21,12 @@ The driver can provide empty directories that are backed by the same filesystem
 Usually, the driver implements all CSI operations itself. When deployed with the `-proxy-endpoint` parameter, it instead proxies all incoming connections for a CSI driver that is [embedded inside the Kubernetes E2E test suite](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/storage/drivers/csi-test) and used for mocking a CSI driver [with callbacks provided by certain tests](https://github.com/kubernetes/kubernetes/blob/5ad79eae2dcbf33df3b35c48ec993d30fbda46dd/test/e2e/storage/csi_mock_volume.go#L110).
 
 ## Deployment
-[Deployment for Kubernetes 1.17 and later](docs/deploy-1.17-and-later.md)
+
+Deployment varies depending on the Kubernetes version your cluster is running:
+- [Deployment for Kubernetes 1.20 and later](docs/deploy-1.20-and-later.md)
+- [Deployment for Kubernetes 1.17 and later](docs/deploy-1.17-and-later.md)
+- [Deployment for Kubernetes 1.16 and earlier](docs/deploy-pre-1.17.md)
+
```
Comment (xing-yang) [inline README.md:29]: @Kartik494 Some of the older K8s release folders are removed recently. Will you be able to update this with K8s 1.21 and later?  Thanks.
https://github.com/kubernetes-csi/csi-driver-host-path/tree/master/deploy

Comment (Kartik494) [reply README.md:29]: @xing-yang Should i only keep deployment section for kubernetes 1.21 and later releases in [deployment](https://github.com/kubernetes-csi/csi-driver-host-path#deployment)  section ? should i remove 1.17 and earlier also?

Comment (xing-yang) [reply README.md:29]: @Kartik494 Yes.

Comment (Kartik494) [reply README.md:29]: @xing-yang Thanks for the clarification , i will update this.

**Comment by xing-yang** [comment]
Hi @Kartik494, can you update this PR?

**Comment by Kartik494** [comment]
@xing-yang  sure , i will addressed the comments ASAP

File: docs/example-snapshots-1.21-and-later.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    Operation:       Update
+    Time:            2022-07-18T04:10:47Z
+  Resource Version:  1112
+  UID:               c82cc073-4ef0-4659-9522-7b2a65df9ff5
+Events:              <none>
+ ```
+
+After having created the `csi-pvc` as described in the deployment validation,
+use the volume snapshot class to dynamically create a volume snapshot:
+  - `$ kubectl apply -f examples/csi-snapshot-v1.yaml`
```
Comment (Kartik494) [inline docs/example-snapshots-1.21-and-later.md:60]: @xing-yang I have added a new yaml file which is using `volumesnapshot` as V1 . As we no longer use v1alpha and v1beta1 , so is this ok to remove those files which are using volumesnapshot as v1alpha1 and v1beta1?

Comment (xing-yang) [reply docs/example-snapshots-1.21-and-later.md:60]: Yes, it is fine to remove those files where v1alpha1 and v1beta1 are used.

Comment (Kartik494) [reply docs/example-snapshots-1.21-and-later.md:60]: Sure i will   addressed the changes .

**Comment by Kartik494** [comment]
@xing-yang added a commit which remove files which are using v1alpha1 and v1beta1 versions and add GA status of VolumeSnapshot in `[docs/example-snapshots-1.21-and-later.md]` file.

File: docs/deploy-1.21-and-later.md [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+
+If no pods are running the snapshot-controller, follow the instructions below to create the snapshot-controller
+
+__Note:__ The above command may not work for clusters running on managed k8s services. In this case, the presence of all VolumeSnapshot CRDs is an indicator that your cluster is ready for hostpath deployment.
+
+### VolumeSnapshot CRDs and snapshot controller installation
+Run the following commands to install these components: 
+```shell
+# Change to the latest supported snapshotter version
+$ SNAPSHOTTER_VERSION=v4.0.1
```
Comment (Kartik494) [inline docs/deploy-1.21-and-later.md:32]: @xing-yang I have used `SNAPSHOTTER_VERSION=v4.0.1` here , should i modify this to latest stable release which is v6.0.1? Please  suggest.
Thanks!!

Comment (xing-yang) [reply docs/deploy-1.21-and-later.md:32]: Yes, please!

Comment (Kartik494) [reply docs/deploy-1.21-and-later.md:32]: Hi @xing-yang when i am trying this with version v6.0.1, i got the following error while deploying

`The CSIDriver "hostpath.csi.k8s.io" is invalid: spec.fsGroupPolicy: Invalid value: "File": field is immutable`

Could you please suggest on this?

Comment (Kartik494) [reply docs/deploy-1.21-and-later.md:32]: the `fsGroupPolicy: File`  field in this [file](https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/deploy/kubernetes-1.21/hostpath/csi-hostpath-driverinfo.yaml) throwing error.

Comment (Kartik494) [reply docs/deploy-1.21-and-later.md:32]: @xing-yang resolved the merge conflict .For now should we used `SNAPSHOTTER_VERSION=v4.0.1` as i am facing error as stated above while using v6.0.1 ?
Thanks!

**Pull request closed without merging**